staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -867,14 +876,14 @@ func (s *store) listPartition(ctx context.Context, client *clientv3.Client, key
 
 		// avoid small allocations for the result slice, since this can be called in many
 		// different contexts and we don't know how significantly the result will be filtered
-		trace.Step("Acquiring ListAppendMux listPartition")
-		listAppendMux.Lock()
-		trace.Step("Acquired ListAppendMux listPartition")
+		// trace.Step("Acquiring ListAppendMux listPartition")
+		// listAppendMux.Lock()
+		// trace.Step("Acquired ListAppendMux listPartition")
 
 		if pred.Empty() {
 			growSlice(v, len(getResp.Kvs))
 		} else {
-			growSlice(v, 2048, len(getResp.Kvs))
+			growSlice(v, 4096, len(getResp.Kvs))
```

Comment:
Community is still using 2048 in 1.18.5. Not sure whether this is helpful. Suggest do not merge into POC branch.

Reply:
this is experimental change trying to reduce the # of list reallocation.